tailormap-components/projects/core/src/lib/user-interface/geometry-confirm-buttons/geometry-confirm-buttons.component.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { GeometryConfirmService } from './geometry-confirm.service';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { ScreenCoordinate } from '../models';
+
+@Component({
+  selector: 'tailormap-geometry-confirm-buttons',
+  templateUrl: './geometry-confirm-buttons.component.html',
+  styleUrls: ['./geometry-confirm-buttons.component.css'],
+})
+export class GeometryConfirmButtonsComponent implements OnInit, OnDestroy {
+
+  private destroyed = new Subject();
+  public position: ScreenCoordinate;
+  public visible = false;
+
+  constructor(private geometryConfirmService: GeometryConfirmService) {
+    this.position = {
+      left: 0,
+      top: 0,
+    };
+    this.geometryConfirmService.positionChanged$
```

Comment:
You could save some lines of code here by not subscribing to 2 observable here but mapping to position and visiblity and using the async pipe in the template. That way you can remove the destroyed subject + OnInit & OnDestroy.

For example: 
`
this.position$ = combineLatest([
      this.geometryConfirmService.positionChanged$,
      this.geometryConfirmService.visibilityChanged$,
    ]).pipe(map(([ position, visible ]) => {
        return visible ? { left: position.left + MARGIN, top: position.top + BUTTON_HEIGHT } : null;
      }));
`

and in the template

`<div class="geometry-confirm" *ngIf="position$ | async as position" [style.left.px]="position.left [style.top.px]="position.top">`

Reply:
Oeh, very nice.